src/Simplify/AstHelpers.elm

```diff
@@ -713,30 +713,19 @@ getTuple expressionNode =
 
 
 getBoolPattern : ModuleNameLookupTable -> Node Pattern -> Maybe Bool
-getBoolPattern lookupTable patternNode =
-    case Node.value patternNode of
-        Pattern.NamedPattern { name } _ ->
-            case name of
-                "True" ->
-                    if ModuleNameLookupTable.moduleNameFor lookupTable patternNode == Just [ "Basics" ] then
-                        Just True
-
-                    else
-                        Nothing
-
-                "False" ->
-                    if ModuleNameLookupTable.moduleNameFor lookupTable patternNode == Just [ "Basics" ] then
-                        Just False
-
-                    else
-                        Nothing
+getBoolPattern lookupTable basePatternNode =
+    case removeParensFromPattern basePatternNode of
+        Node variantPatternRange (Pattern.NamedPattern variantPattern _) ->
+            case ( ModuleNameLookupTable.moduleNameAt lookupTable variantPatternRange, variantPattern.name ) of
+                ( Just [ "Basics" ], "True" ) ->
+                    Just True
+
+                ( Just [ "Basics" ], "False" ) ->
+                    Just False
```

Comment:
I agree that this is easier to read, but the reason I did it the way it was before was to avoid computing the module name when `variantPattern.name` was not `True` or `False`. Since in practice this function will be rarely called with a node where `variantPattern.name` is one of those two, I think this is worth keeping as it was before for performance reasons.

Reply:
Ok, reverted.

This to me strongly smells to me like something a smart compiler could automatically optimize (checking already evaluated things first).

I guess elm's really strong eager eval always constructs the tuple, even if we reversed the order so that option also doesn't exist sadg

Reply:
Yeah, this is a benefit of laziness that we don't have. Not sure if it would apply in this specific case though :thinking: